packages/runtime/src/modules/AttributeListenerModule.ts

```diff
@@ -19,20 +21,69 @@ export class AttributeListenerModule extends RuntimeModule {
         if (createdAttribute.content["@type"] === "RelationshipAttribute" && createdAttribute.content.confidentiality === RelationshipAttributeConfidentiality.Private) return;
+        if (await AttributeListenerModule.relationshipAttributeOfNonActiveRelationship(services, createdAttribute)) return;
 
         const getAttributeListenersResult = await services.consumptionServices.attributeListeners.getAttributeListeners();
         if (getAttributeListenersResult.isError) {
-            this.logger.error("Could not get attribute listeners", getAttributeListenersResult.error);
+            this.logger.error("Could not get LocalAttributeListeners", getAttributeListenersResult.error);
             return;
         }
 
         const attributeListeners = getAttributeListenersResult.value;
         const promises = attributeListeners.map((attributeListener) =>
-            this.createRequestIfAttributeMatchesQuery(services, attributeListener, event.data, event.eventTargetAddress)
+            this.createRequestIfAttributeMatchesQuery(services, attributeListener, createdAttribute, event.eventTargetAddress)
         );
         await Promise.all(promises);
     }
 
+    private static async relationshipAttributeOfNonActiveRelationship(services: RuntimeServices, attribute: LocalAttributeDTO): Promise<boolean> {
+        if (attribute.content["@type"] !== "RelationshipAttribute") {
+            return false;
+        }
+
+        const activeRelationshipToPeer = (
+            await services.transportServices.relationships.getRelationships({ query: { peer: attribute.shareInfo?.peer, status: RelationshipStatus.Active } })
+        ).value;
+
+        if (activeRelationshipToPeer.length === 0) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private async handleRelationshipChanged(event: RelationshipChangedEvent) {
+        const changedRelationship = event.data;
+        if (changedRelationship.status !== RelationshipStatus.Active || changedRelationship.auditLog.length > 2) return;
+
+        const lastAuditLogEntry = changedRelationship.auditLog[changedRelationship.auditLog.length - 1];
+        if (lastAuditLogEntry.reason === RelationshipAuditLogEntryReason.AcceptanceOfCreation) {
+            const services = await this.runtime.getServices(event.eventTargetAddress);
+            const relationshipAttributesWithPeer = (
+                await services.consumptionServices.attributes.getAttributes({ query: { "content.@type": "RelationshipAttribute", "shareInfo.peer": changedRelationship.peer } })
+            ).value;
+
+            for (const relationshipAttribute of relationshipAttributesWithPeer) {
+                if (relationshipAttribute.content["@type"] !== "RelationshipAttribute") return;
+                if (relationshipAttribute.content.confidentiality === RelationshipAttributeConfidentiality.Private) return;
```

Comment:
Removing the type check unfortunately leads to the `confidentiality` of the `relationshipAttribute.content` being underlined in red because it is allegedly not guaranteed that the `confidentiality` property is set for `relationshipAttribute.content`, although the query was previously specified so that `relationshipAttribute.content` is guaranteed to be a RelationshipAttribute... 😿😂 

I'd probably leave the confidentiality check where it is to comply with the above pattern.